src/chk/chk_internal.h

```diff
@@ -40,8 +46,119 @@ enum chk_rpc_opc {
 
 #undef X
 
-int chk_rejoin(void);
+/* dkey for check DB under sys_db */
+#define CHK_DB_TABLE		"chk"
 
-int chk_pause(void);
+/* akey for leader bookmark under CHK_DB_TABLE */
+#define CHK_BK_LEADER		"leader"
+
+/* akey for engine bookmark under CHK_DB_TABLE */
+#define CHK_BK_ENGINE		"engine"
+
+/* akey for check property under CHK_DB_TABLE */
+#define CHK_PROPERTY		"property"
+
+#define CHK_BK_MAGIC_LEADER	0xe6f703da
+#define CHK_BK_MAGIC_ENGINE	0xe6f703db
+#define CHK_BK_MAGIC_POOL	0xe6f703dc
+
+/*
+ * XXX: Please be careful when change CHK__CHECK_INCONSIST_CLASS__CIC_UNKNOWN
+ *	to avoid hole is the struct chk_property.
+ */
+#define CHK_POLICY_MAX		(CHK__CHECK_INCONSIST_CLASS__CIC_UNKNOWN + 1)
+#define CHK_POOLS_MAX		(1 << 6)
+#define CHK_RANKS_BITMAP_MAX	(1 << 24)
+
+/*
+ * Each check instance has a unique leader engine that uses key "chk/leader" under its local
+ * sys_db to trace the check instance.
+ *
+ * For each engine, include the leader engine, there is a system level key "chk/engine" under
+ * the engine's local sys_db to trace the check instance on the engine. When server (re)start
+ * the check module uses it to determain whether needs to rejoin the check instance.
+ *
+ * For each pool, there is a key "chk/$pool_uuid" under the engine's local sys_db to trace
+ * check process for the pool on related engine.
+ */
+struct chk_bookmark {
+	uint32_t			cb_magic;
+	uint32_t			cb_version;
+	uint64_t			cb_gen;
+	Chk__CheckScanPhase		cb_phase;
```

Comment:
Is this generated by protobuf? Why do we need to use it here?

Reply:
Yes, it is generated by protobuf. We use it here to indicate this field is for some kind of scan phase.